packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1050,21 +1063,148 @@ class RenderParagraph extends RenderBox
             final TextRange originalRange = infoAttribute.range;
             attributes.add(
               infoAttribute.copy(
-                  range: TextRange(start: offset + originalRange.start,
-                      end: offset + originalRange.end)
+                range: TextRange(
+                  start: offset + originalRange.start,
+                  end: offset + originalRange.end,
+                ),
               ),
             );
           }
           buffer.write(label);
           offset += label.length;
         }
-        _cachedAttributedLabel = AttributedString(buffer.toString(), attributes: attributes);
+        _cachedAttributedLabels = <AttributedString>[AttributedString(buffer.toString(), attributes: attributes)];
       }
-      config.attributedLabel = _cachedAttributedLabel!;
+      config.attributedLabel = _cachedAttributedLabels![0];
       config.textDirection = textDirection;
     }
   }
 
+  ChildSemanticsConfigurationsResult _childSemanticsConfigurationsDelegate(List<SemanticsConfiguration> childConfigs) {
+    if (_cachedAttributedLabels == null) {
+      final ChildSemanticsConfigurationsResult result = _buildChildSemanticsConfigurationsResultAndFillCache(childConfigs);
+      assert(() {
+        // Verify the cache will produce the same result.
+        final ChildSemanticsConfigurationsResult resultBuiltFromCache = _buildChildSemanticsConfigurationsResultAndFillCache(childConfigs);
```

Comment:
Should this be calling _buildChildSemanticsConfigurationsResultFromCache instead?